uPortal-security/uPortal-security-mvc/src/test/java/org/apereo/portal/url/MaxInactiveFilterTest.java

```diff
@@ -0,0 +1,277 @@
+
+        final MaxInactiveFilter filter = new MaxInactiveFilter();
+        ReflectionTestUtils.setField(filter, "personManager", personManager);
+        ReflectionTestUtils.setField(filter, "maxInactiveStrategy", maxInactiveStrategy);
+
+        filter.doFilter(req, resp, chain);
+
+        verify(person, times(1)).getSecurityContext();
+        verify(person, times(1)).getAttribute(IPerson.USERNAME);
+        verify(personManager, times(1)).getPerson(req);
+        verifyZeroInteractions(maxInactiveStrategy);
+        verifyZeroInteractions(resp);
+        verifyZeroInteractions(session);
+        verify(chain, only()).doFilter(req, resp);
+    }
+
+    @Test
+    public void noPersonWorkflow() throws IOException, ServletException {
+        final HttpSession session = mock(HttpSession.class);
+        final HttpServletRequest req = mock(HttpServletRequest.class);
+        when(req.getSession(false)).thenReturn(session);
+        final ServletResponse resp = mock(ServletResponse.class); // no calls, used in doFilter()
+        final FilterChain chain = mock(FilterChain.class);
+
+        final IPersonManager personManager = mock(IPersonManager.class);
+        when(personManager.getPerson(req)).thenReturn(null);
+
+        final IMaxInactiveStrategy maxInactiveStrategy = mock(IMaxInactiveStrategy.class);
+
+        final MaxInactiveFilter filter = new MaxInactiveFilter();
+        ReflectionTestUtils.setField(filter, "personManager", personManager);
+        ReflectionTestUtils.setField(filter, "maxInactiveStrategy", maxInactiveStrategy);
+
+        filter.doFilter(req, resp, chain);
+
+        verify(personManager, times(1)).getPerson(req);
+        verifyZeroInteractions(maxInactiveStrategy);
+        verifyZeroInteractions(resp);
+        verifyZeroInteractions(session);
+        verify(chain, only()).doFilter(req, resp);
+    }
+
+    @Test
+    public void noSessionWorkflow() throws IOException, ServletException {
+        final HttpServletRequest req = mock(HttpServletRequest.class);
+        when(req.getSession()).thenReturn(null);
+        final ServletResponse resp = mock(ServletResponse.class); // no calls, used in doFilter()
+        final FilterChain chain = mock(FilterChain.class);
+
+        final IPersonManager personManager = mock(IPersonManager.class);
+
+        final IMaxInactiveStrategy maxInactiveStrategy = mock(IMaxInactiveStrategy.class);
+
+        final MaxInactiveFilter filter = new MaxInactiveFilter();
+        ReflectionTestUtils.setField(filter, "personManager", personManager);
+        ReflectionTestUtils.setField(filter, "maxInactiveStrategy", maxInactiveStrategy);
+
+        filter.doFilter(req, resp, chain);
+
+        verifyZeroInteractions(personManager);
+        verifyZeroInteractions(maxInactiveStrategy);
+        verifyZeroInteractions(resp);
+        verify(req, only()).getSession(false);
+        verify(chain, only()).doFilter(req, resp);
+    }
+}
```

Comment:
:+1:  nice to see all the test coverage.

Reply:
:-)